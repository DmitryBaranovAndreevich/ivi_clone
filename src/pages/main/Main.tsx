import CategoriesSlider from '../../components/categoriesSlider/CategoriesSlider';
import Slider from '../../components/mainSlider/MainSlider';
import CardFilm from '../../components/cardFilm/cardFilm';
import styles from './Main.module.scss';
import example from './../../assests/example.jpg';
import { Outlet } from 'react-router-dom';

const Main = () => {
  return (
    <div className={styles.container}>
      <h2>MainPage</h2>
      <Slider />
      {/* <div className={styles.wrapper}>
        <CategoriesSlider title={'Рекомендую посмотреть >'} size={'medium'} />
        <CategoriesSlider title={'(TOP 10) за неделю'} size={'big'} />
      </div> */}
      <CardFilm
        image={example}
        name="qwertyqqqqqqqfrgjhkl;jhvghbjnkml,"
        year={2022}
        country="qwerty"
        genre="qwerty"
      />
      <Outlet />
    </div>
  );
};

export default Main;
