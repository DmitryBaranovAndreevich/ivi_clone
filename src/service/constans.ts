export const mediaQuery = {
  $smallDesktop: '(max-width: 1160px)',
  $laptop: '(max-width: 850px)',
  $tablets: '(max-width: 550px)',
  $phones: '(max-width: 450px)',
};

export const URL = 'http://localhost:3000/login';
export const headers = {
  Accept: 'application/json',
};

export const URL_TO_FILMS = 'http://localhost:3000/api';
