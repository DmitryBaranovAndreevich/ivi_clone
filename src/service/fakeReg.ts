function fakeRequest(time: number) {
  return new Promise((res, rej) => {
    setTimeout(res, time);
  });
}

function fakeLogin(user: { email: string; password: string }) {
  return fakeRequest(2000).then(() => ({ accesstoken: 'acces', refreshtoken: 'refresh' }));
}

export default fakeLogin;
