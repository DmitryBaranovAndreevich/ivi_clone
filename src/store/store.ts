import reviewReducer from './reducers/ReviewSlice';
import moviesSortReduser from './reducers/MoviesSort';
import { appApi } from './api/appApi';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import userLoginReduser from './reducers/UserLoginSlice';
import appReducer from './reducers/App';
import moviesFilterReduser from './reducers/MoviesFilter';
import { filmApi } from './api/filmApi';
import { personApi } from './api/personApi';
import { reviewApi } from './api/reviewApi';

const rootReducer = combineReducers({
  appReducer,
  userLoginReduser,
  moviesSortReduser,
  reviewReducer,
  moviesFilterReduser,
  [appApi.reducerPath]: appApi.reducer,
  [filmApi.reducerPath]: filmApi.reducer,
  [personApi.reducerPath]: personApi.reducer,
  [reviewApi.reducerPath]: reviewApi.reducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat([appApi.middleware, filmApi.middleware, personApi.middleware]),
  });
};

export type TRootState = ReturnType<typeof rootReducer>;
export type TAppStore = ReturnType<typeof setupStore>;
export type TAppDispatch = TAppStore['dispatch'];
