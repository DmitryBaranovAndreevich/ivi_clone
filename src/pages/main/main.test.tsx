import 'whatwg-fetch';
import renderer from 'react-test-renderer';
import Main from './Main';
import { Provider } from 'react-redux';
import { store } from '../../store/mainStore';
import { URL } from '../../service/constans';
import fetchMock from 'jest-fetch-mock';
import { useTranslation } from 'react-i18next';
jest.mock('react-i18next', () => ({
  useTranslation: jest.fn(),
}));
jest.useFakeTimers();
describe('Тестим главную страницу', () => {
  beforeEach(() => {
    (useTranslation as jest.Mock).mockReturnValue({ t: (key: string) => key });
  });

  afterAll(() => {
    jest.runAllTimers();
  });

  it('Главная страница рендерится без ошибок', () => {
    jest.spyOn(Storage.prototype, 'getItem');
    localStorage.getItem = jest.fn();
    jest.runAllTimers();

    const tree = renderer.create(
      <Provider store={store}>
        <Main />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
