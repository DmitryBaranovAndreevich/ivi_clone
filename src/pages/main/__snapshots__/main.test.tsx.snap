// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим главную страницу Главная страница рендерится без ошибок 1`] = `
<div>
  <div
    className="{spinnerStyle.spinner} undefined"
  >
    <i
      className="undefined undefined"
    />
  </div>
</div>
`;
